rust-runtime/aws-smithy-runtime-api/src/client/retries/classifiers.rs

```diff
@@ -0,0 +1,206 @@
+enum Inner {
+    // The default priority for the `HttpStatusCodeClassifier`.
+    HttpStatusCodeClassifier,
+    // The default priority for the `ModeledAsRetryableClassifier`.
+    ModeledAsRetryableClassifier,
+    // The default priority for the `TransientErrorClassifier`.
+    TransientErrorClassifier,
+    // The priority of some other classifier.
+    Other(i8),
+}
+
+impl PartialOrd for RetryClassifierPriority {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        Some(other.as_i8().cmp(&self.as_i8()))
+    }
+}
+
+impl Ord for RetryClassifierPriority {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        other.as_i8().cmp(&self.as_i8())
+    }
+}
+
+impl RetryClassifierPriority {
+    /// Create a new `RetryClassifierPriority` with the default priority for the `HttpStatusCodeClassifier`.
+    pub fn http_status_code_classifier() -> Self {
+        Self {
+            inner: Inner::HttpStatusCodeClassifier,
+        }
+    }
+
+    /// Create a new `RetryClassifierPriority` with the default priority for the `ModeledAsRetryableClassifier`.
+    pub fn modeled_as_retryable_classifier() -> Self {
+        Self {
+            inner: Inner::ModeledAsRetryableClassifier,
+        }
+    }
+
+    /// Create a new `RetryClassifierPriority` with the default priority for the `TransientErrorClassifier`.
+    pub fn transient_error_classifier() -> Self {
+        Self {
+            inner: Inner::TransientErrorClassifier,
+        }
+    }
+
+    /// Create a new `RetryClassifierPriority` with lower priority than the given priority.
+    pub fn with_lower_priority_than(other: Self) -> Self {
+        Self {
+            inner: Inner::Other(other.as_i8() + 1),
+        }
+    }
+
+    /// Create a new `RetryClassifierPriority` with higher priority than the given priority.
+    pub fn with_higher_priority_than(other: Self) -> Self {
+        Self {
+            inner: Inner::Other(other.as_i8() - 1),
+        }
+    }
+
+    fn as_i8(&self) -> i8 {
+        match self.inner {
+            Inner::HttpStatusCodeClassifier => 0,
+            Inner::ModeledAsRetryableClassifier => 10,
+            Inner::TransientErrorClassifier => 20,
+            Inner::Other(i) => i,
+        }
+    }
+}
+
+impl Default for RetryClassifierPriority {
+    fn default() -> Self {
+        Self {
+            inner: Inner::Other(0),
+        }
+    }
+}
+
+/// Classifies what kind of retry is needed for a given [`InterceptorContext`].
+pub trait ClassifyRetry: Send + Sync + fmt::Debug {
+    /// Run this classifier on the [`InterceptorContext`] to determine if the previous request
+    /// should be retried. Returns a [`RetryAction`].
+    fn classify_retry(&self, ctx: &InterceptorContext) -> RetryAction;
+
+    /// The name of this retry classifier.
+    ///
+    /// Used for debugging purposes.
+    fn name(&self) -> &'static str;
+
+    /// The priority of this retry classifier. Classifiers with a higher priority will run before
```

Comment:
rather than noting the order it should probably say that higher priority overrides lower priority